### PR TITLE
 Refactor/#8 Commit Template Update

### LINKED ISSUE #8: Create commit template
Add commit template to ensure that each commit contains the issue number as prefix. The agreed upon format for commit message is `I#${issue_number} ${commit_message}`.

### PR DESCRIPTION
**Pull request by SayefReyadh**
Fixes #8

Resolves #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [ ] New feature
- [x] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
The previous `I#${issue_number} ${commit_message}` format was not tagging the issue, so refactored it to `I-#${issue_number} ${commit_message}` 

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [x] I have added or updated necessary documentation to reflect the changes.
- [x] I have followed the project's coding style and guidelines.
- [x] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [x] I have checked for any potential conflicts with other branches or pull requests.

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: .githooks/commit-msg [hunk 1]
```diff
@@ -7,12 +7,12 @@ TEMPORARY_FILE_PATH=$1
...
 ## init regex to match commit-message format
-COMMIT_MSG_REGEX="\I#[0-9]+|Merge branch\s.*"
+COMMIT_MSG_REGEX="\I-#[0-9]+|Merge branch\s.*"
 
 ## checking commit-message format
 if ! [[ $COMMIT_MSG =~ $COMMIT_MSG_REGEX ]]; then
     echo -e "Your commit-message format is not valid:\n$COMMIT_MSG\n"
-    echo "Did you add an 'I' before the #issueNumber? Valid format example:"
+    echo "Did you add an 'I-' before the #issueNumber? Valid format example:"
     echo "I#issueNumber My commit message"
```
Comment (shuaibvm) [inline .githooks/commit-msg:16]: Please consider adding an `I-` in this example line as well

**Review by shuaibvm — APPROVED** [review]
Looks good! Thank you for the proactive step :D

**Review by shuaibq3 — APPROVED** [review]
Looks good! Thank you for the proactive step :D

File: .githooks/commit-msg (hunk 1 above)
Comment (shuaibq3) [reply .githooks/commit-msg:16]: Please consider adding an I- in this example line as well

**Review by shuaibq3 — APPROVED** [review]
LGTM

**Pull request merged**